src/components/ManageAccountsDrawer/components/ImportAccounts.tsx

```diff
@@ -246,18 +247,25 @@ export const ImportAccounts = ({ chainId, onClose }: ImportAccountsProps) => {
     })
   }, [])
 
-  // TODO: Loading state
   const handleDone = useCallback(async () => {
-    // for every new account that is active, fetch the account and upsert it into the redux state
-    for (const accountId of toggledAccountIds) {
-      const isEnabled = selectIsAccountIdEnabled(store.getState(), { accountId })
-      if (isEnabled) {
-        continue
-      }
-      await dispatch(portfolioApi.endpoints.getAccount.initiate({ accountId, upsertOnFetch: true }))
-    }
+    setIsSubmitting(true)
+
+    // For every new account that is active, fetch the account and upsert it into the redux state
+    await Promise.all(
+      Array.from(toggledAccountIds).map(async accountId => {
+        const isEnabled = selectIsAccountIdEnabled(store.getState(), { accountId })
+        if (isEnabled) {
+          return
+        }
+        await dispatch(
+          portfolioApi.endpoints.getAccount.initiate({ accountId, upsertOnFetch: true }),
+        )
+      }),
+    )
 
     if (!accounts) {
```

Comment:
Do we want to also add `accounts` as a criteria for `isDisabled` on l.366 to match this? If this should never be reached, then we should never let the user reach this without `accounts` in the first place?

Reply:
Also feels like we may want to move this (now paranoia if guarding enabled state against `!accounts`) check above - if there are no initial accounts, the list would be empty, and all the above would be a no-op since `toggleAccountIds` would itself be empty